### DISCUSSION TITLE
elem_side_builder.h missing in moose-libmesh

### DISCUSSION THREAD

**Discussion by WulfHans** [idaholab/moose#20122] [Q&A General] [answered]
I updated my moose environment today and the previously working project failed building with:
```
moose_projects/moose/framework/build/header_symlinks/Assembly.h:24:10: fatal error: libmesh/elem_side_builder.h: No such file or directory
   24 | #include "libmesh/elem_side_builder.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

The versions installed in the moose environment are reported as up-to-date as well:
```
$ conda list
# packages in environment at /home/hans/miniconda3/envs/moose:
#
# Name                    Version                   Build  Channel
moose-libmesh             2021.10.27              build_0    idaholab
moose-libmesh-vtk         6.3.0                   build_7    idaholab
```
I inspected the header in the environment and indeed, this header file is missing, whereas all the other elem* header are present:
```
~/miniconda3/envs/moose/libmesh/include/libmesh$ ls elem*.h
elem_assembly.h  elem_cutter.h  elem.h  elem_hash.h  elem_internal.h  elem_quality.h  elem_range.h
```

According to the docs, for instance
https://mooseframework.inl.gov/docs/doxygen/libmesh/files.html
it should be there. I also tried reupdating everything, reinstalling libmesh, nothing helped.

Please help me to fix this. Thanks in advance!

**Comment by GiudGiud** [comment]
Hello 
2021.10.27 is too old, the conda update failed.
The easiest thing to do is to wipe out the environment and re-run the conda commands to re-create it

guillaume

**Reply by WulfHans** [reply]
Well it does not say so:

```
conda update --all
Collecting package metadata (current_repodata.json): done
Solving environment: done

# All requested packages already installed.
```

But it might be not completely correct here...

**Reply by GiudGiud** [reply]
Yeah the dependency resolution does not allow a newer update at this point 
It thinks it’s up to date 

this is why you ll have to wipe it

**Comment by milljm** [comment, marked as answer]
I just saw @GiudGiud post. But I am going to continue posting mine...

You are also using our older idaholab channel. The new channel is:

```bash
conda config --add channels https://conda.software.inl.gov/public
```

Be sure to remove the old one:
```bash
conda config --remove channels idaholab
```

This older channel will someday be removed.
Also, you may wish to upgrade to mamba. It is a much quicker Conda package installer.


Also.... the inability to upgrade is probably due to the fact a python version in moose-tools is preventing it. You should probably wipe the entire 'moose' environment, and reinstall, for the least amount of headaches.

**Reply by WulfHans** [reply]
Wiping the whole environment and upgrading to mamba solved it. Thanks for the fast and efficient help!

**Comment by loganharbour** [comment]
FYI - not sure who added it, but adding `[Solved]` to the title prefix isn't necessary. Marking as "answered" is sufficient!
